src/app-layer-ssl.c

```diff
@@ -143,11 +143,97 @@ static int SSLv3ParseHandshakeType(SSLState *ssl_state, uint8_t *input,
     switch (ssl_state->curr_connp->handshake_type) {
         case SSLV3_HS_CLIENT_HELLO:
             ssl_state->flags |= SSL_AL_FLAG_STATE_CLIENT_HELLO;
+
+            /* skip version */
+            initial_input += 2;
+
+            /* skip random */
+            initial_input += 32;
+
+            /* skip session id */
+            uint8_t session_id_length = *(initial_input++);
+            initial_input += session_id_length;
+
+            if ((initial_input - input) > input_len)
+                goto end;
+
+            /* skip cipher suites */
+            uint16_t cipher_suites_length = *(uint16_t *)initial_input;
+            initial_input += ntohs(cipher_suites_length) + 2;
+
+            if ((initial_input - input) > input_len)
+                goto end;
+
+            /* skip compression methods */
+            uint8_t compression_methods_length = *(initial_input++);
+            initial_input += compression_methods_length;
+
+            if ((initial_input - input) > input_len)
+                goto end;
+
+            uint16_t extensions_len = *(uint16_t *)initial_input;
+            initial_input += 2;
+
+            if ((initial_input - input) > input_len)
+                goto end;
+
+            /* return if no extensions are found */
+            if (ntohs(extensions_len) <= 0)
+                goto end;
+
+            uint16_t processed_len = 0;
+            while (processed_len < ntohs(extensions_len))
+            {
+                uint16_t ext_type = *(uint16_t *)initial_input;
+                initial_input += 2;
+
+                if ((initial_input - input) > input_len)
+                    goto end;
+
+                uint16_t ext_len = *(uint16_t *)initial_input;
+                initial_input += 2;
+
+                if ((initial_input + ntohs(ext_len) - input) > input_len)
+                    goto end;
+
+                switch (ntohs(ext_type)) {
```

Comment:
Think I'd prefer doing the ntohs when assigning the value to ext_type a few lines up.